src/main/java/macrobase/transform/FeatureTransform.java

```diff
@@ -0,0 +1,39 @@
+package macrobase.transform;
+
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+
+import javax.xml.crypto.Data;
+import java.util.Iterator;
+import java.util.function.Consumer;
+
+public class FeatureTransform implements Iterator<Datum> {
+
+    DataIngester ingester;
+
+    public FeatureTransform(MacroBaseConf conf, DataIngester ingester) {
+        // TODO: apply specified configurations
+        this.ingester = ingester;
+    }
+
+    /**
+      * Applies the transformation this class represents to the datum given and
+      * outputs a new transformed datum.
+      * @param datumToTransform datum that should be transformed
+      * @return new Datum that has transformation applied to it.
+      */
+    public Datum transform(Datum datumToTransform) {
```

Comment:
If you subclass `FeatureTransform` you can override a protected method. The main reason is to make sure clients don't call transform manually and instead use the public facing iterator APIs. (It's partly because I was wondering why transform existed if `FeatureTransform implements Iterator` until I saw the implementation of next. Although that could also be somewhat cleared up by just moving next to be above transform :smile:).